feat(dashboard): Invalid variable pill improvements

BiswaViraj opened the pull request:
### What changed? Why was the change needed?

<!-- Also include any relevant links, such as Linear tickets, Slack discussions, or design documents. -->

### Screenshots

<!-- If the changes are visual, include screenshots or screencasts. -->

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Related enterprise PR

<!-- A link to a dependent pull request  -->

### Special notes for your reviewer

<!-- Specific instructions or considerations you want to highlight for the reviewer. -->

</details>

BiswaViraj committed:
feat(dashboard): enhance liquid autocomplete validation for dynamic paths

BiswaViraj committed:
feat(dashboard): implement dynamic path suggestions in autocomplete and enhance variable filtering

BiswaViraj committed:
feat(dashboard): add invalid dynamic path values for enhanced autocomplete validation

ChmaraX commented on apps/dashboard/src/components/workflow-editor/steps/email/maily.tsx:
```diff
@@ -204,7 +205,29 @@ const dedupAndSortVariables = (
+        required: false,
+      };
+    });
+  }
```
Please rebase on next and move this to `variables.ts` file that now contains this logic

SokratisVidros committed:
fix(dashboard): Fix merge conflicts

SokratisVidros committed:
fix(dashboard): Exclude naked payload from suggested variables

The new UX prepends payload to all unknown {{foo}} variables as the user types them.

SokratisVidros committed:
WIP

SokratisVidros committed:
fixup! WIP

SokratisVidros commented:
Closing this one in favor of https://github.com/novuhq/novu/pull/8259
